Code/Framework/AzCore/AzCore/Math/LineSegment.h

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Math/Vector3.h>
+#include <AzCore/RTTI/TypeInfoSimple.h>
+
+namespace O3DE
+{
+    class LineSegment
+    {
+    public:
+        AZ_TYPE_INFO(LineSegment, "{7557da1e-cc20-11ec-9d64-0242ac120002}");
+
+        explicit LineSegment(const AZ::Vector3& start, const AZ::Vector3& end);
+
+        const AZ::Vector3& GetStart() const;
+        const AZ::Vector3& GetEnd() const;
+        
+        LineSegment& operator=(const LineSegment& rhs);
+        bool operator==(const LineSegment& rhs) const;
+        bool operator!=(const LineSegment& rhs) const;
+
+        //TODO: implement
+        //!
+        float GetLength();
+    
+        //!
+        AZ::Vector3 GetDirection();
+        
+        //!
+        AZ::Vector3 GetPoint(float t);
```

Comment:
Is this meant to some kind of interpolate function?
I would argument that returning a Vector3 isn't a "point"(perhaps we should have a Point3 class?)

Reply:
Since I suggested this one... :) Yeah, GetPoint() would basically be an interpolate, there are multiple places with line segments where you end up doing point = start + (end - start) * t, so it seems like a reasonable convenience method to expose.

That's a really good point that Vector3 _isn't_ a point, it's a vector, but we currently use it interchangeably as a holding bin for anything requiring XYZ components - it's a position in the Transform class, it's a dimension size in the BoxShape class, it's a set of 3 angles in TransformBus::SetLocalRotation, etc. I could see the merits of having different types for the different use cases, but if we do anything more complex than a bunch of "using" aliases, we might run into a bunch of extraneous type conversions or duplicated implementations.

Reply:
I personally have tried using a `Point` class in a math library before (my own little hobby project) and while it actually seems like it might be quite nice (at least in theory from a mathematical/linear algebra perspective - there's an interesting conversation about it here - https://www.reedbeta.com/blog/on-vector-math-libraries/) in practice it's just a pain and ends up causing a lot more friction than you'd like (especially when interacting with other math types). I propose we don't get into adding a `Point` class as part of this change if we can help it 😛